pytorch_lightning/loggers/mlflow.py

```diff
@@ -21,21 +21,25 @@
 from time import time
 from typing import Any, Dict, Optional, Union
 
-try:
-    import mlflow
-    from mlflow.tracking import MlflowClient
-except ModuleNotFoundError:  # pragma: no-cover
-    mlflow = None
-    MlflowClient = None
-
 
 from pytorch_lightning import _logger as log
 from pytorch_lightning.loggers.base import LightningLoggerBase, rank_zero_experiment
-from pytorch_lightning.utilities import rank_zero_only, rank_zero_warn
+from pytorch_lightning.utilities import rank_zero_only, rank_zero_warn, _module_available
+
 
 LOCAL_FILE_URI_PREFIX = "file:"
 
 
+_MLFLOW_AVAILABLE = _module_available("mlflow")
+try:
+    import mlflow
+    from mlflow.tracking import MlflowClient
+# todo: there seems to be still some remaining import error with Conda env
+except ImportError:
+    _MLFLOW_AVAILABLE = False
+    mlflow, MlflowClient = None, None
```

Comment:
```suggestion
_MLFLOW_CLIENT_AVAILABLE = _module_available("mlflow.tracking.MlflowClient")
if _MLFLOW_CLIENT_AVAILABLE:
    import mlflow
    from mlflow.tracking import MlflowClient
else:
    mlflow, MlflowClient = None, None
```

Reply:
why testing `MlflowClient` what is the case that it is missing?

Reply:
Because MLFlow without its client can't be used. I find wierd to use the clever _module_available and add an ImportError below with try / except.

Reply:
then we shall set min lib that has not to allow user to install the version that is incompatible... so this is not the place